packages/core/src/browser/dialogs.ts

```diff
@@ -24,6 +24,33 @@ export class DialogProps {
     readonly title: string;
 }
 
+export type DialogMode = 'open' | 'preview';
+
+export type DialogError = string | boolean | {
+    message: string
```

Comment:
why not `readonly`?

Reply:
have not thought about it, not sure from what it is going to protect:
- the only client is `AbstractDialog` and it does not mutate it
- mutating by subclasses in `isValid` method looks fine if someone want to custoize an error message